virtual_rainforest/models/animals/model.py

```diff
@@ -0,0 +1,118 @@
+"""The :mod:`~virtual_rainforest.animal.model` module creates a
+:class:`~virtual_rainforest.animal.model.AnimalModel` class as a child of the
+:class:`~virtual_rainforest.core.model.BaseModel` class. At present a lot of the
+abstract methods of the parent class (e.g.
+:func:`~virtual_rainforest.core.model.BaseModel.setup` and
+:func:`~virtual_rainforest.core.model.BaseModel.spinup`) are overwritten using
+placeholder functions that don't do anything. This will change as the
+:mod:`virtual_rainforest` model develops. The factory method
+:func:`~virtual_rainforest.animals.model.AnimalModel.from_config` exists in a more
+complete state, and unpacks a small number of parameters from our currently pretty
+minimal configuration dictionary. These parameters are then used to generate a class
+instance. If errors crop here when converting the information from the config dictionary
+to the required types (e.g. :class:`~numpy.timedelta64`) they are caught and then
+logged, and at the end of the unpacking an error is thrown. This error should be
+caught and handled by downstream functions so that all model configuration failures
+can be reported as one.
+"""  # noqa: D205, D415
+
+
+from __future__ import annotations
+
+from typing import Any
+
+import pint
+from numpy import datetime64, timedelta64
+
+from virtual_rainforest.core.logger import LOGGER
+from virtual_rainforest.core.model import BaseModel, InitialisationError
+
+
+class AnimalModel(BaseModel):
+    """A class describing the animal model.
+
+    Describes the specific functions and attributes that the animal module should
+    possess. Currently it is incomplete and mostly just a copy of the template set out
+    in AnimalModel.
+
+    Args:
+        update_interval: Time to wait between updates of the model state.
+
```

Comment:
Probably should get rid of this long gap after the last Argument

Reply:
I'm slightly surprised that flake8-docstring doesn't pick up on this tbh.

Reply:
It does do some odd things - there was an issue with it allowing incomplete `Args` when there were other sections? `Raises` in particular?